airbyte-integrations/connectors/source-pinterest/source_pinterest/manifest.yaml

```diff
@@ -0,0 +1,3190 @@
+    type: DefaultErrorHandler
+    backoff_strategies:
+      - type: WaitTimeFromHeader
+        header: "X-RateLimit-Reset"
+    response_filters:
+      - type: HttpResponseFilter
+        action: IGNORE
+        error_message: "Max rate limit exceeded"
+        predicate: "{{ response.get('code') == 8 }}"
+      - type: HttpResponseFilter
+        action: IGNORE
+        error_message: "Can't access catalog product groups because there is no existing catalog."
+        http_codes: [409]
+      - type: HttpResponseFilter
+        action: FAIL
+        error_message: "Try to re-authenticate because current refresh token is not valid."
+        error_message_contains: "401 Client Error: Unauthorized for url"
+        http_codes: [401]
+      - type: HttpResponseFilter
+        action: RETRY
+        http_codes: [500]
+
+  analytics_error_handler:
+    type: DefaultErrorHandler
+    backoff_strategies:
+      - type: ConstantBackoffStrategy
+        backoff_time_in_seconds: 1
+    response_filters:
+      - type: HttpResponseFilter
+        action: RETRY
+        http_codes: [400]
+
+  # Requester
+  requester:
+    type: HttpRequester
+    url_base: "https://api.pinterest.com/v5/"
+    http_method: GET
+    authenticator: "#/definitions/authenticator"
+    error_handler: "#/definitions/base_error_handler"
+
+  # Selector
+  selector:
+    type: RecordSelector
+    extractor:
+      type: DpathExtractor
+      field_path: ["items"]
+
+  # Retrievers
+  retriever:
+    type: SimpleRetriever
+    record_selector: "#/definitions/selector"
+    requester: "#/definitions/requester"
+    paginator:
+      type: DefaultPaginator
+      pagination_strategy:
+        type: CursorPagination
+        cursor_value: "{{ response.get('bookmark') }}"
+      page_token_option:
+        type: RequestOption
+        field_name: bookmark
+        inject_into: request_parameter
+
+  board_based_retriever:
+    $ref: "#/definitions/retriever"
+    partition_router:
+      type: SubstreamPartitionRouter
+      parent_stream_configs:
+        - type: ParentStreamConfig
+          parent_key: "id"
+          stream: "#/definitions/boards_stream"
+          partition_field: "id"
+
+  ad_account_based_retriever:
+    $ref: "#/definitions/retriever"
+    partition_router:
+      type: SubstreamPartitionRouter
+      parent_stream_configs:
+        - type: ParentStreamConfig
+          parent_key: "id"
+          stream: "#/definitions/ad_accounts_stream"
+          partition_field: "id"
+
+  analytics_retriever:
+    $ref: "#/definitions/ad_account_based_retriever"
+    record_selector:
+      type: RecordSelector
+      extractor:
+        type: DpathExtractor
+        field_path: []
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "#/definitions/requester"
+      error_handler:
+        type: CompositeErrorHandler
+        error_handlers:
+          - "#/definitions/base_error_handler"
+          - "#/definitions/analytics_error_handler"
```

Comment:
might be a nit since the error handlers look to be mutually exclusive, but I think the `analytics_error_handler` should come first. If we look at the original stream implementation, we apply the 400 error check first, then default back to the other status code after

Reply:
Updated